covalent_awslambda_plugin/awslambda.py

```diff
@@ -178,28 +182,27 @@ async def submit_task(
         """
         loop = asyncio.get_running_loop()
         fut = loop.run_in_executor(
-            None, self.submit_task_sync, function_name, func_filename, result_filename
+            None,
+            self.submit_task_sync,
+            function_name,
+            func_filename,
+            result_filename,
+            exception_filename,
         )
         return await fut
 
-    def get_status_sync(self, object_key: str):
+    def get_status_sync(self, object_key: str) -> bool:
         with self.get_session() as session:
-            while not self._key_exists:
-                s3_client = session.client("s3")
-                try:
-                    current_keys = [
-                        item["Key"]
-                        for item in s3_client.list_objects(Bucket=self.s3_bucket_name)["Contents"]
-                    ]
-                    self._key_exists = object_key in current_keys
-
-                    if not self._key_exists:
-                        time.sleep(0.5)
-
-                    return self._key_exists
-                except botocore.exceptions.ClientError as ce:
-                    app_log.exception(ce)
-                    raise
+            s3_client = session.client("s3")
+            try:
+                current_keys = [
+                    item["Key"]
+                    for item in s3_client.list_objects(Bucket=self.s3_bucket_name)["Contents"]
```

Comment:
Two problems with `list_objects`:
1) It only returns [up to 1000 objects](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.list_objects), whereas the user's s3 bucket could easily contain more than that.
2) Iterating through all objects in the bucket will slow down as more object are added to the bucket.

For both of these reasons, it would better to query the object metadata directly using [head_object()](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.head_object).

Reply:
updated!